beacon-chain/db/filesystem/blob.go

```diff
@@ -177,3 +185,75 @@ func (p blobNamer) partPath() string {
 func (p blobNamer) path() string {
 	return p.fname(sszExt)
 }
+
+// Prune prunes blobs in the base directory based on the retention epoch.
+// It deletes blobs older than currentEpoch - (retentionEpoch+bufferEpochs).
+// This is so that we keep a slight buffer and blobs are deleted after n+2 epochs.
+func (bs *BlobStorage) Prune(cliCtx *cli.Context, currentSlot primitives.Slot) error {
```

Comment:
I still think that having the cli.Context as an argument to the blob storage is a code smell.

If you don't want the caller to figure out the retention slot, can you initialize BlobStorage with the flag value?